### DISCUSSION TITLE
Call Reach API from node.js backend application

### DISCUSSION THREAD

**Discussion by lorenzopolidori** [reach-sh/reach-lang#1334] [Q&A] [answered]
I am working on a frontend application with a node.js backend that would need to connect to an admin wallet and call a Reach API to fetch information from an Algorand smart contract.

What is the recommended approach to do this? Shall I use `stdlib.connectAccount(networkAccount) => Promise<acc>` from the node.js backend, using wallet secrets encrypted at rest in the backend?

I would like to avoid having to rely on the end user to connect to their wallet and make the Reach API call from the frontend, so that the network fees would be paid by the admin wallet, rather than the end user, as this information is not related to the user, but rather general information about the current state of the application.

**Comment by jeapostrophe** [comment, marked as answer]
If you are going to be sending publications, then you need to have a real account with money in it. You should not share your secrets with your users. So, they need to use their own account, or you need to provide a traditional Web2 frontend where your secrets are known.

If you are just observing the chain, then you just need a Reach account abstraction, so you should use `stdlib.createAccount` --- https://docs.reach.sh/frontend/#js_createAccount --- to get one. It will have no money, but that doesn't matter, because you won't be sending any transactions.

**Reply by lorenzopolidori** [reply]
Thanks, I would only be observing the chain through a Reach API implemented as part of a `parallelReduce`. The API returns loop variables.

Is there not a small fee to pay to call the API, even if it is only to read?

**Reply by jeapostrophe** [reply]
You should replace the API with a View, then it will be callable by accounts with no funds.

**Reply by lorenzopolidori** [reply]
I have tried to use `stdlib.createAccount` to connect to a deployed contract on Algorand Testnet:
```js
const account = await stdlib.createAccount()
const sppContract = account.contract(backend, contractInfo)
```
but I got this error message:
```
Error: WR44: verifyContract failed: failed to lookup application ({application_id}): Error: Request has been terminated
```
Connecting works fine with a real account. Could this be that on Algorand an account must have a non-zero balance to do anything, even to connect to a contract?

**Reply by jeapostrophe** [reply]
No, it really doesn't need any balance to do that. I think the "request termination" is a different issue, potentially related to the node you're connecting to
